pieces.py

```diff
@@ -3,24 +3,65 @@ def __init__(self, color: str, i: int, j: int):
         self.color = color
         self.position = (i, j)
         self.firstMove = True
+    def valid_moves(self, board): #Returns a list of valid coordinates for respective piece
+        pass
+    def insideBoard(self, i: int, j: int) -> bool:
+        if square[0] < 8 and suare[1] < 8:
+            return True
+        return False
 
 class Pawn(Piece):
-    def valid_moves(self, board):
-    
+    def valid_moves(self, board: list[list[Piece]]):
+        i, j = self.position  
+        coordinates = []          
+        if self.color == 'w':
+            #Check front moves
+            if self.firstMove and not board[i + 2][j]:
+                coordinates.append((i + 2, j))
+            if self.insideBoard((i + 1, j)) and not board[i + 1][j]:
+                coordinates.append((i + 1, j))
+            #Check left diagonal
+            if self.insideBoard(i + 1, j - 1):
+                if not board[i + 1][j - 1] or board[i + 1][j - 1].color == 'b':
+                    coordinates.append((i + 1, j - 1))
+            #Check right diagonal
+            if self.insideBoard(i + 1, j + 1):
+                if not board[i + 1][j + 1] or board[i + 1][j + 1].color == 'b':
+                    coordinates.append((i + 1, j - 1))
+        else:
+            #Check front moves
+            if self.firstMove and not board[i - 2][j]:
+                coordinates.append((i - 2, j))
+            if self.insideBoard((i - 1, j)) and not board[i - 1][j]:
+                coordinates.append((i - 1, j))
+            #Check left diagonal
+            if self.insideBoard(i - 1, j + 1):
+                if not board[i - 1][j + 1] or board[i - 1][j + 1].color == 'w':
+                    coordinates.append((i - 1, j + 1))
+            #Check right diagonal
+            if self.insideBoard(i - 1, j - 1):
+                if not board[i - 1][j - 1] or board[i - 1][j - 1].color == 'w':
+                    coordinates.append((i - 1, j - 1))
+        return coordinates
```

Comment:
same comment as in https://github.com/Omarcito017/chess/pull/6#discussion_r535506003: this is the same logic yet its repeated twice for each color. There should be a way that we only right this once without the outer if/else check. 

This may be through may be moving the logic of white or black into a different function (for example, maybe  `move_foward()` will then hold the logic of whether it is white or black to be + or - 1). Or maybe it would be better that we rotate the board around so that the logic is the same. Or maybe you can do some math logic inside each function (unsure how this would work). Probably a lot of other solutions.